File: experiments/runs/2026-09-22_00-02-08_claim-office-python-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/catalogue.py
"""The MHPCO price list."""

from errors import ScenarioError

MAIN_ITEM_BASE_PREMIUMS = {
    "sword": 100,
    "amulet": 60,
    "staff": 80,
    "potion": 40,
}

MAIN_ITEM_INSURANCE_VALUES = {
    "sword": 1000,
    "amulet": 600,
    "staff": 800,
    "potion": 400,
}

COMPONENT_TYPES = ("rune", "moonstone")
COMPONENT_BASE_PREMIUM = 25
COMPONENT_INSURANCE_VALUE = 250


def is_component(item_type):
    return item_type in COMPONENT_TYPES


def is_known_type(item_type):
    return is_component(item_type) or item_type in MAIN_ITEM_BASE_PREMIUMS


def require_known_type(item_type):
    if not is_known_type(item_type):
        raise ScenarioError(f"unknown item type: {item_type!r}")


def insurance_value(item_type):
    require_known_type(item_type)
    if is_component(item_type):
        return COMPONENT_INSURANCE_VALUE
    return MAIN_ITEM_INSURANCE_VALUES[item_type]

File: experiments/runs/2026-09-23_07-01-38_claim-office-example-mapping_exact-ptdd-v1-cc_opus-5-5-no-thinking/src/catalogue.ts
import type { Item } from "./claimOffice.js";

interface PriceListEntry {
  insuranceValue: number;
  basePremium: number;
}

const MAIN_ITEMS: Record<string, PriceListEntry> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
};

export const COMPONENT_TYPES = ["rune", "moonstone"];
export const COMPONENT: PriceListEntry = { insuranceValue: 250, basePremium: 25 };

export function isComponent(item: Item): boolean {
  return COMPONENT_TYPES.includes(item.type);
}

export function mainItemEntry(item: Item): PriceListEntry {
  const entry = MAIN_ITEMS[item.type];
  if (entry === undefined) {
    throw new Error(`Unknown item type: ${item.type}`);
  }
  return entry;
}

export function insuranceValue(item: Item): number {
  return isComponent(item) ? COMPONENT.insuranceValue : mainItemEntry(item).insuranceValue;
}
